tests/unit/test_coordinate_utilities.py

```diff
@@ -0,0 +1,431 @@
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(
+                    self.varinfo, '/Soil_Moisture_Retrieval_Data_AM/surface_flag'
+                ),
+                expected_override_dimensions_AM,
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions for the longitude variable'
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(self.varinfo, self.longitude),
+                expected_override_dimensions_AM,
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions for the latitude variable'
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(self.varinfo, self.latitude),
+                expected_override_dimensions_AM,
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions science variable with a different grid'
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(
+                    self.varinfo, '/Soil_Moisture_Retrieval_Data_PM/surface_flag_pm'
+                ),
+                expected_override_dimensions_PM,
+            )
+
+    def test_get_row_col_sizes_from_coordinate_datasets(self):
+        """Ensure that the correct row and column sizes are
+        returned for the requested coordinates
+        """
+        expected_row_col_sizes = (5, 10)
+        lat_1d_array = np.array([1, 2, 3, 4])
+        lon_1d_array = np.array([6, 7, 8, 9])
+        lat_irregular_array = np.array([[1, 2, 3], [3, 4, 5]])
+        lon_irregular_array = np.array([[6, 7], [8, 9], [10, 11]])
+        with self.subTest('Retrieves the expected row col sizes from the coordinates'):
+            self.assertEqual(
+                get_row_col_sizes_from_coordinate_datasets(self.lat_arr, self.lon_arr),
+                expected_row_col_sizes,
+            )
+        with self.subTest('Retrieves the expected row col sizes for the 1d array'):
+            self.assertEqual(
+                get_row_col_sizes_from_coordinate_datasets(lat_1d_array, lon_1d_array),
+                (4, 4),
+            )
+
+        with self.subTest(
+            'Raises an exception when the lat and lon array shapes do not match'
+        ):
+            with self.assertRaises(IrregularCoordinateVariables) as context:
+                get_row_col_sizes_from_coordinate_datasets(
+                    lat_irregular_array, lon_irregular_array
+                )
+            self.assertEqual(
+                context.exception.message,
+                f'Longitude coordinate shape: "{lon_irregular_array.shape}"'
+                f'does not match the latitude coordinate shape: "{lat_irregular_array.shape}"',
+            )
+
+    def test_get_valid_indices(self):
+        """Ensure that latitude and longitude values are correctly identified as
+        ascending or descending.
+
+        """
```

Comment:
I think there are a couple of other test cases here:

* I would _definitely_ like to see a test that shows if you have fill values specified, that the specific latitude or longitude checks aren't also done. Preferably there should be two tests, one for latitude and one for longitude, as they have separate branches through the conditional block. If this isn't what you want to be happening, then these tests can help you fix up the function. Tangible examples of what I mean:
  * Fill Value give, longitude array has a value of 361 degrees east. That value is included in the output.
  * Fill Value given, latitude array has a value of -100 degrees north. That value is included in the output.
* A test with no fill values, but filters out-of-range latitude values (both `latitude < -90` and `latitude > 90` degrees north)
* A test with no fill values, but filters out-of-range longitude values (both `longitude < -180` and `longitude > 180` degrees east).

Reply:
updated - 51b110c7294791d63551dc14d8ada49573eafeb0